plugins/embed-optimizer/hooks.php

```diff
@@ -23,90 +53,156 @@
+			// iterating over the tags of the embed markup alone as is passed into the embed_oembed_html filter.
+			if ( $html_processor instanceof OD_HTML_Tag_Processor ) {
+				if ( 'FIGURE' === $html_processor->get_tag() ) {
+					if ( $html_processor->is_tag_closer() ) {
+						--$figure_depth;
+						if ( $figure_depth <= 0 ) {
+							// We reached the end of the embed.
+							break;
+						}
+					} else {
+						++$figure_depth;
+						// Move to next element to start looking for IFRAME or SCRIPT tag.
+						continue;
+					}
+				}
+				if ( 0 === $figure_depth ) {
+					continue;
 				}
 			}
-		} elseif ( 'SCRIPT' === $html_processor->get_tag() ) {
-			if ( ! $html_processor->get_attribute( 'src' ) ) {
-				$has_inline_script = true;
-			} else {
-				++$script_count;
-				if ( ! $html_processor->set_bookmark( 'script' ) ) {
-					wp_trigger_error( __FUNCTION__, esc_html__( 'Embed Optimizer unable to set script bookmark.', 'embed-optimizer' ) );
-					return $html;
+
+			if ( 'IFRAME' === $html_processor->get_tag() ) {
+				$loading_value = $html_processor->get_attribute( 'loading' );
+				if ( empty( $loading_value ) ) {
+					++$iframe_count;
+					if ( ! $html_processor->set_bookmark( $bookmark_names['iframe'] ) ) {
+						throw new Exception( __( 'Embed Optimizer unable to set iframe bookmark.', 'embed-optimizer' ) );
+					}
+				}
+			} elseif ( 'SCRIPT' === $html_processor->get_tag() ) {
+				if ( ! $html_processor->get_attribute( 'src' ) ) {
+					$has_inline_script = true;
+				} else {
+					++$script_count;
+					if ( ! $html_processor->set_bookmark( $bookmark_names['script'] ) ) {
+						throw new Exception( __( 'Embed Optimizer unable to set script bookmark.', 'embed-optimizer' ) );
+					}
 				}
 			}
-		}
-	}
-	// If there was only one non-inline script, make it lazy.
-	if ( 1 === $script_count && ! $has_inline_script && $html_processor->has_bookmark( 'script' ) ) {
-		add_action( 'wp_footer', 'embed_optimizer_lazy_load_scripts' );
-		if ( $html_processor->seek( 'script' ) ) {
-			if ( $html_processor->get_attribute( 'type' ) ) {
-				$html_processor->set_attribute( 'data-original-type', $html_processor->get_attribute( 'type' ) );
+		} while ( $html_processor->next_tag() );
+		// If there was only one non-inline script, make it lazy.
+		if ( 1 === $script_count && ! $has_inline_script && $html_processor->has_bookmark( $bookmark_names['script'] ) ) {
+			$needs_lazy_script = true;
+			if ( $html_processor->seek( $bookmark_names['script'] ) ) {
+				if ( $html_processor->get_attribute( 'type' ) ) {
+					$html_processor->set_attribute( 'data-original-type', $html_processor->get_attribute( 'type' ) );
+				}
+				$html_processor->set_attribute( 'type', 'application/vnd.embed-optimizer.javascript' );
+			} else {
+				wp_trigger_error( __FUNCTION__, esc_html__( 'Embed Optimizer unable to seek to script bookmark.', 'embed-optimizer' ) );
```

Comment:
Ditto here for using a placeholder for `script`

Reply:
Done in d8ac531